onlinejudge/service/yukicoder.py

```diff
@@ -350,13 +350,12 @@ def from_url(cls, url: str) -> Optional['YukicoderProblem']:
             try:
                 n = int(basename)
             except ValueError:
-                pass
+                return None
```

Comment:
I prefer to return early.
If not, code reader have to think "what happened if n is None?" and read until the end.

Reply:
Another option for return here.
```Python
        if result.scheme in ('', 'http', 'https') \
                and result.netloc == 'yukicoder.me':
            try:
                n = int(basename)
                if dirname == '/problems/no':
                    return cls(problem_no=int(n))
                if dirname == '/problems':
                    return cls(problem_id=int(n))
            except ValueError:
                pass
        return None
```

Reply:
And n never become None, we can remove if branch.

Reply:
The early return is entirely reasonable in this case, but I would like to avoid it for consistency.

All other methods named as `from_url` are written in the style like `if (pattern A) { return a } ; if (pattern B) { return b } ; if (pattern C) { return c } ; return None`. This style is required one since patterns of URL may be added in the future. So I want this `from_url` method to use this style too.
Example:
https://github.com/kmyk/online-judge-tools/blob/8ba1f3fef36288c0b9960dfd52563859dc2ff6ab/onlinejudge/service/atcoder.py#L812-L836
https://github.com/kmyk/online-judge-tools/blob/8ba1f3fef36288c0b9960dfd52563859dc2ff6ab/onlinejudge/service/atcoder.py#L1049-L1083